code/src/main/java/com/onlinecontacttracing/servlets/authentication/CheckForApiAuthorizationServlet.java

```
@@ -0,0 +1,128 @@
+
+  abstract void useCredential(Credential credential);
+  // should return "check-for-api-authorization" for now
+  abstract String getServletURIName();
+  //Update the userId with the newly created credential
+  abstract void updateUser(String userId);
+
+  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+  private static final List<String> SCOPES = Collections.singletonList("https://www.googleapis.com/auth/contacts.readonly");
+  private static final String CREDENTIALS_FILE_PATH = "WEB-INF/credentials.json";
+  private static final String url = "https://8080-ac896ae1-5f0c-45b5-8dfe-19fa4d3d8699.us-east1.cloudshell.dev";
+  static final Logger log = Logger.getLogger(CheckForApiAuthorizationServlet.class.getName());
+
+  //Creates the user's credential
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get flow for token response
+    GoogleAuthorizationCodeFlow flow = getFlow(response);
+    if (flow != null) {
+      String code = request.getParameter("code");
+      String idTokenString = request.getParameter("state");
+
+      // Get userId
+      String userId = getUserId(idTokenString, flow, response);
+      if (userId != null) {
+        TokenResponse tokenResponse = flow.newTokenRequest(code).setRedirectUri(url+getServletURIName()).execute();
+        Credential credential = flow.createAndStoreCredential(tokenResponse, userId);
+        updateUser(userId);
+        useCredential(credential);
+      }
+    }
+  }
+  
+  //Creates the url for authorizing the user
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get token to pass into redirect
+    String idToken = request.getParameter("idToken");
+    // Get flow to build url redirect
+    GoogleAuthorizationCodeFlow flow = getFlow();
+
+    if (flow != null) {
```

So, at this point are we just failing silently?